Main_Control_Node/src/XBee/XBee.c

```diff
@@ -0,0 +1,112 @@
+#include "XBee.h"
+
+/** Initializes the STM32 UART on GPIO B10 and B11 to a baudrate of 9600
+
+**/
+void uart_init(void)
+{
+
+	// Enable GPIOA clock, AFIO
+	RCC->APB2ENR |= 0x9UL;			//Enables Port B Clock and AFIO Clock
+	GPIOB->CRH &= ~(0xFF00UL);	//B11: Floating Input
+	GPIOB->CRH |= 0x4B00UL;			//B10: Push-Pull, Output 50MHZ
+	
+	RCC->APB1ENR |= (0x1UL << 18);		//Enable USART 3 CLock
+	
+	USART3->CR2 = 0;									
+	USART3->CR3 = 0;									
+	USART3->CR1 |= 0xC;								//Transmitter and Receiver Enable
+
+	// By Default, set baudrate to 9600
+	USART3->BRR = 0xEA6UL;
+
+	// Enable USART3
+	USART3->CR1 |= 0x1 << 13;
+}
+
+/** Transmits a CAN message via the previously initialized STM32 UART
+		
+Passes: An instance of the can_msg_t data type
+**/
+void XBeeTransmitCan(can_msg_t* msg_tx)
+{
+	//Timestamp: 4 bytes
+	uint8_t time[2];
+	uint8_t count;
+	
+	time[0] = 0xFFUL & (RTC->CNTH);
+	time[1] = 0xFFUL & (RTC->CNTL);
+	
+	for (count = 0; count < 2; count++)
+	{
+		XBeeSendChar(time[count]);
+	}
+
+	//CAN ID: 2 bytes

```

Comment:
remove these if not needed (or leave comment about why this should still be here and commented)